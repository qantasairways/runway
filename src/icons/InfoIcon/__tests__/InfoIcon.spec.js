import React from 'react';
import { shallow } from 'enzyme';
import shortid from 'shortid';

import InfoIcon from '..';

jest.mock('shortid');

describe('InfoIcon', () => {
  it('should render', () => {
    shortid.generate.mockReturnValue('1a2b3c');
    const component = shallow(
      <InfoIcon width="36px" height="36px" className="my-favourite-icon" />
    );

    expect(component).toMatchSnapshot();
  });

  it('should set the width', () => {
    const component = shallow(<InfoIcon width="36px" />);
    const result = component.find('svg').props().width;

    expect(result).toEqual('36px');
  });

  it('should set the height', () => {
    const component = shallow(<InfoIcon height="36px" />);
    const result = component.find('svg').props().height;

    expect(result).toEqual('36px');
  });

  it('should set the class name', () => {
    const component = shallow(<InfoIcon className="my-favourite-icon" />);
    const result = component.find('svg').props().className;

    expect(result).toEqual('my-favourite-icon');
  });
});
