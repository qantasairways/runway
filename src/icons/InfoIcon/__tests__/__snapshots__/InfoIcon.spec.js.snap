// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoIcon should render 1`] = `
<svg
  className="my-favourite-icon"
  focusable="false"
  height="36px"
  viewBox="0 0 16 16"
  width="36px"
>
  <g
    stroke="none"
    strokeWidth="1"
  >
    <g
      transform="translate(-969.000000, -184.000000)"
    >
      <g
        transform="translate(515.000000, 126.000000)"
      >
        <g
          transform="translate(0.000000, 57.000000)"
        >
          <path
            d="M462,1 C457.581667,1 454,4.582 454,9 C454,13.4186667 457.581667,17 462,17 C466.418,17 470,13.4186667 470,9 C470,4.582 466.418,1 462,1 L462,1 Z M462,15.6666667 C458.318193,15.6666667 455.333333,12.6821277 455.333333,9.00032044 C455.333333,5.31851318 458.318193,2.33333333 462,2.33333333 C465.681807,2.33333333 468.666667,5.31851318 468.666667,9.00032044 C468.666667,12.6821277 465.681807,15.6666667 462,15.6666667 L462,15.6666667 Z M463.830361,10.8597581 C463.631709,10.8597581 462.85212,11.9427256 462.42717,11.9427256 C462.313849,11.9427256 462.257531,11.853763 462.257531,11.7644792 C462.257531,11.5608607 462.413175,11.2419441 462.498165,11.0508511 L463.518389,8.56696285 C464.028671,7.33112087 463.376739,7 462.767472,7 C461.945558,7 461.208636,7.36933947 460.642034,7.85333476 C460.457377,8.01937694 460,8.66202762 460,8.90579167 C460,8.98704635 460.086355,9.08211112 460.187388,9.08211112 C460.446454,9.08211112 460.953324,7.84080934 461.4776,7.84080934 C461.591262,7.84080934 461.718918,7.95578632 461.619592,8.18445563 L460.627699,10.426828 C460.528373,10.6436142 460.046762,11.3864683 460.046762,11.997966 C460.046762,12.4819612 460.400717,13 460.910999,13 C462.342521,13 464,11.4205117 464,11.0508511 C464,10.9361953 463.901015,10.8597581 463.830361,10.8597581 L463.830361,10.8597581 Z M463.086124,5 C462.503349,5 462,5.48620102 462,6.08074957 C462,6.6350937 462.357895,7 462.900797,7 C463.496651,7 464,6.5407155 464,5.91925043 C464,5.36456559 463.615949,5 463.086124,5 L463.086124,5 Z"
            id="1a2b3c"
          />
        </g>
      </g>
    </g>
  </g>
</svg>
`;
